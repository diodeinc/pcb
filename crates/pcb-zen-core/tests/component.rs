#[macro_use]
mod common;

snapshot_eval!(component_properties, {
    "C146731.kicad_sym" => include_str!("resources/C146731.kicad_sym"),
    "test_props.zen" => r#"
        # Import component factory from current directory.
        load(".", MyComponent = "C146731")

        # Instantiate with pin connections and a custom property.
        MyComponent(
            name = "U1",
            pins = {
                "ICLK": Net("ICLK"),
                "Q1": Net("Q1"),
                "Q2": Net("Q2"),
                "Q3": Net("Q3"),
                "Q4": Net("Q4"),
                "GND": Net("GND"),
                "VDD": Net("VDD"),
                "OE": Net("OE"),
            },
            footprint = "SMD:0805",
            properties = {"CustomProp": "Value123"},
        )
    "#
});

snapshot_eval!(interface_net_incompatible, {
    "Module.zen" => r#"
        SingleNet = interface(signal = Net)
        signal_if = SingleNet(name="sig")

        Component(
            name = "test_comp",
            footprint = "test_footprint",
            pin_defs = {"in": "1", "out": "2"},
            pins = {
                "in": signal_if,  # This should fail - interfaces not accepted for pins
                "out": Net()
            }
        )
    "#
});
