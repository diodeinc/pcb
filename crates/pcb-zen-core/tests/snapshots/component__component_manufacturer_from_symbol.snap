---
source: crates/pcb-zen-core/tests/component.rs
expression: output
---
Module {
    name: "<root>",
    source: "test.zen",
    children: [
        FrozenValue(
            Component {
                name: "test_comp",
                manufacturer: "ACME Corp",
                footprint: "test_footprint",
                prefix: "U",
                connections: {
                    "GND": FrozenValue(
                        Net {
                            name: "GND",
                            id: "<ID>",
                            symbol: FrozenValue(
                                NoneType,
                            ),
                        },
                    ),
                    "VCC": FrozenValue(
                        Net {
                            name: "VCC",
                            id: "<ID>",
                            symbol: FrozenValue(
                                NoneType,
                            ),
                        },
                    ),
                },
                properties: {
                    "symbol_name": FrozenValue(
                        "TestSymbol",
                    ),
                    "symbol_path": FrozenValue(
                        "/test_symbol.kicad_sym",
                    ),
                },
                symbol: FrozenValue(
                    Symbol {
                        name: Some(
                            "TestSymbol",
                        ),
                        pins: {
                            "1": "VCC",
                            "2": "GND",
                        },
                        properties: {
                            "Manufacturer_Name": "ACME Corp",
                            "Reference": "U",
                        },
                    },
                ),
            },
        ),
    ],
}
[]
